File: DemoServer/AdditionalLanguages/go/RelatedDocuments/loadRelatedDocuments/LoadRelatedDocuments.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    loadRelatedDocuments(5.2, "050-123456")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

type Product struct {
    Name string
    Supplier string
    Category string
    PricePerUnit float64
}

type Supplier struct {
    ID string
    Name string
    Phone string
}

//region Demo
func loadRelatedDocuments(pricePerUnit float64, phone string) error {

    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()
    
    //region Step_1
    var product *Product
    err = session.Include("supplier").
                  Load(&product, "products/34-A")
    if err != nil {
        return err
    }
    if product == nil {
        return nil
    }
    //endregion

    //region Step_2
    var supplier *Supplier
    err = session.Load(&supplier, product.Supplier)
    if err != nil || supplier == nil {
        return err
    }
    //endregion

    //region Step_3
    product.PricePerUnit = pricePerUnit
    supplier.Phone = phone
    //endregion

    //region Step_4
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    //endregion

    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/relatedDocuments/loadRelatedDocuments/LoadRelatedDocuments.java
package net.ravendb.demo.relatedDocuments.loadRelatedDocuments;

//region Usings
import net.ravendb.client.documents.session.IDocumentSession;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;
import net.ravendb.demo.common.models.Product;
import net.ravendb.demo.common.models.Supplier;

public class LoadRelatedDocuments {

    public void run(RunParams runParams) {
        double pricePerUnit = runParams.getPricePerUnit();
        String phone = runParams.getPhone();

        //region Demo
        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
            //region Step_1
            Product product = session
                .include("Supplier")
                .load(Product.class, "products/34-A");
            //endregion

            //region Step_2
            Supplier supplier = session.load(Supplier.class, product.getSupplier());
            //endregion

            //region Step_3
            product.setPricePerUnit(pricePerUnit);
            supplier.setPhone(phone);
            //endregion

            //region Step_4
            session.saveChanges();
            //endregion
        }
        //endregion
    }

    public static class RunParams {
        private double pricePerUnit;
        private String phone;

        public double getPricePerUnit() {
            return pricePerUnit;
        }

        public void setPricePerUnit(double pricePerUnit) {
            this.pricePerUnit = pricePerUnit;
        }

        public String getPhone() {
            return phone;
        }

        public void setPhone(String phone) {
            this.phone = phone;
        }
    }
}
